docs/proposals/0008-ToolAuthAPI.md

```diff
@@ -0,0 +1,459 @@
+# Tool Authorization in Agentic Networking
+
+This proposal defines authorization policies for tool access from AI agents running inside a Kubernetes cluster to MCP servers running in the Kubernetes cluster or outside of the Kubernetes cluster. By default, an AI agent can call initialize, notifications/initialized and tools/list. To enforce a "zero trust" security posture, a tools/call is denied unless it is allowed through the Tool Auth API described in this proposal.
+
+# Non-Goals
+
+The authentication of MCP tool access is not within the scope of this proposal, and will be explored separately in the future.
+
+# Use Cases & Motivation 
+
+## Personas
+
+*AI Engineer*: A hands-on builder focused on the end-to-end development, deployment, and optimization of AI agents. They are distinct from ML Researchers and ML Engineers; AI Engineers are product-first, operating on the other side of the LLM Inference Serving API, and are not responsible for training, tuning, or deploying the models themselves.
+
+*AI Platform Engineer:* A builder and operator of the foundational platform that enables AI engineers to develop and deploy agents at scale. 
+
+*Tool Developer:* A builder focused on developing MCP tools that can be leveraged by agents.
+
+*AI Security Engineer:* A specialist focused on designing automated safeguards to ensure AI agents operate safely and securely, while sometimes also wearing multiple hats (e.g., policy architect, risk manager, and compliance advisor) ensuring security is integrated across all aspects of AI development and deployment.
+
+## User Journeys
+
+#### Agent Identity 
+
+As an AI Engineer, I want to assign a unique, verifiable identity to my agent running in Kubernetes, so that gateways or external systems can securely authenticate it and make authorization decisions.
+
+#### Protocol-Aware Authorization for MCP Tools
+
+As an AI Engineer, I want to create authorization policies to specify which individual tools (e.g., getWeather, sendEmail) my agent is permitted to call on an allow-listed MCP server, so that I can enforce least-privilege access at the specific tool-function level, not just the network endpoint.
```

Comment:
Following on from a comment discussion in the original google doc, 
how do you feel about a user journey about filtering the list of tools returned from a tool list so that tokens and time are not wasted trying to call a tool that a user doesn't have access to?